src/drive/mobile/actions/settings.js

```diff
@@ -46,80 +46,10 @@ export const setBackupContacts = backupContacts => ({
 })
 export const setTokenScope = scope => ({ type: TOKEN_SCOPE, scope })
 
-// errors
-
-export const wrongAddressErrorMsg =
-  'mobile.onboarding.server_selection.wrong_address'
-export const wrongAddressWithEmailErrorMsg =
-  'mobile.onboarding.server_selection.wrong_address_with_email'
-export const wrongAddressV2ErrorMsg =
-  'mobile.onboarding.server_selection.wrong_address_v2'
-export const wrongAddressError = () => ({
-  type: ERROR,
-  error: wrongAddressErrorMsg
-})
-export const wrongAddressWithEmailError = () => ({
-  type: ERROR,
-  error: wrongAddressWithEmailErrorMsg
-})
-export const wrongAddressV2Error = () => ({
-  type: ERROR,
-  error: wrongAddressV2ErrorMsg
-})
-export class OnBoardingError extends Error {
-  constructor(message) {
-    super(message)
-    this.name = 'OnBoardingError'
-  }
-}
-
-// registration
-
-export const registerDevice = cozyClient => async (dispatch, getState) => {
-  let url
-  try {
-    url = checkURL(getState().mobile.settings.serverUrl)
-    await dispatch(setUrl(url))
-  } catch (err) {
-    if (err.name === MAIL_EXCEPTION) {
-      dispatch(wrongAddressWithEmailError())
-    } else {
-      dispatch(wrongAddressError())
-    }
-    throw err
-  }
-
-  let isV2Instance
-  try {
-    isV2Instance = await cozyClient.isV2(url)
-  } catch (err) {
-    // this can happen if the HTTP request to check the instance version fails;
-    // in that case, it is likely to fail again and be caught during the authorize process,
-    // which is designed to handle this
-    isV2Instance = false
-  }
-
-  if (isV2Instance) {
-    dispatch(wrongAddressV2Error())
-    throw new Error('The cozy address entered is a V2 instance')
-  }
-
-  return cozyClient
-    .register(getState().mobile.settings.serverUrl)
-    .then(({ client, token }) => {
-      dispatch(setClient(client))
-      dispatch(setTokenScope(token.scope))
-      startReplication(dispatch, getState)
-    })
-    .catch(err => {
-      if (err.message === 'REGISTRATION_ABORT') {
-        cozy.client._storage.clear()
-      } else {
-        dispatch(wrongAddressError())
-        logException(err)
-      }
-      throw err
-    })
+export const saveCredentials = (client, token) => (dispatch, getState) => {
```

Comment:
This action creator is only called [here](https://github.com/cozy/cozy-drive/pull/530/files#diff-8278b5d865d4e4819d89588ae0ea7788R47), I wonder if we should just dispatch the individual actions.